### PR TITLE
Update the logic of a `record.Key` formatting

### PR DESCRIPTION
**Pull request by voscob**
### Description

I've changed the logic of a `record.Key` formatting.

The `keyColumns` is an optional field.
If the field is not empty the system uses it to format `record.Key`, if not - makes a request to the database and uses the
received list of primary keys of the specified table. If the table does not contain primary keys, the system uses the
value of the `orderingColumn` field as the `keyColumns` value.

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-snowflake/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,9 +20,9 @@ The config passed to `Configure` can contain the following fields.
...
-| `orderingColumn` | The name of a column that the connector will use for ordering rows. Its values must be unique and suitable for sorting, otherwise, the snapshot won't work correctly.                                                                           | yes      | "id"                                                   |
-| `snapshot`       |  Whether or not the plugin will take a snapshot of the entire table before starting cdc mode, by default true.                                                                                                                                  | no       | "false"                                                |
+| `primaryKeys`    | Column names that records should use for their `Key` fields.                                                                                                                                                                                    | no       | "id,name"                                              |
```
Comment (maksenius) [inline README.md:23]: Did you check that snowflake supports composite primary keys?

Comment (voscob) [reply README.md:23]: Yes, of course:
https://docs.snowflake.com/en/sql-reference/sql/show-primary-keys.html

> Lists primary keys for the specified table, or for all tables in the current or specified schema, or for all tables in the current or specified database, or for all tables in the current account.

**Review by maksenius — APPROVED** [review]

File: config/config.go [hunk 2]
```diff
@@ -77,9 +76,8 @@ func Parse(cfg map[string]string) (Config, error) {
 		config.Columns = strings.Split(strings.ToUpper(colsRaw), ",")
 	}
 
-	// Columns in snowflake is uppercase.
```
Comment (hariso) [inline config/config.go:80]: I believe this comment is especially useful now that we called strings.ToUpper below.

Comment (maksenius) [reply config/config.go:80]: yep, I returned comment

File: acceptance_test.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright © 2022 Meroxa, Inc.
+// Copyright © 2023 Meroxa, Inc.
```
Comment (hariso) [inline acceptance_test.go:1]: It turned out that this year should be the year of the first publication of the work, i.e. 2022 in this case. The linter, however, is always checking if the year is the current year. In Conduit, we changed the linter check: https://github.com/ConduitIO/conduit/pull/803

Comment (maksenius) [reply acceptance_test.go]: I changed it, Thanks for the this information

**Review by hariso — APPROVED** [review]
:rocket:

**Pull request merged**